waspc/data/Generator/templates/react-app/src/actions/index.js

```diff
@@ -27,6 +29,18 @@ function makeOptimisticUpdatesConfig(optimisticUpdatesConfig) {
   }))
 }
 
+function makeOptimisticUpdateMutationFn(actionFn, optimisticUpdatesConfig) {
+  return function optimisticallyUpdateQueries(item) {
```

Comment:
Lambdas are very common in these cases, yes.

I still consider a regular function expression a better choice, mainly because I can name it (i.e., it won't be anonymous in stack traces, making it easier to troubleshoot and debug). There are other (minor and possibly subjective advantages), full list here: #487.